wpiunits/src/main/java/edu/wpi/first/units/Measure.java

```diff
@@ -135,16 +135,27 @@ default Measure<U> divide(double divisor) {
   }
 
   /**
-   * Divides this measurement by some constant divisor and returns the result. This is equivalent to
-   * {@code divide(divisor.baseUnitMagnitude())}
+   * Divides this measurement by another measure and performs some dimensional analysis to reduce
+   * the units.
    *
-   * @param divisor the dimensionless measure to divide by
+   * @param <U2> the type of the other measure to multiply by
+   * @param other the unit to multiply by
    * @return the resulting measure
-   * @see #divide(double)
-   * @see #times(double)
    */
-  default Measure<U> divide(Measure<Dimensionless> divisor) {
-    return divide(divisor.baseUnitMagnitude());
+  default <U2 extends Unit<U2>> Measure<?> divide(Measure<U2> other) {
+    if (unit().m_baseType.equals(other.unit().m_baseType)
```

Comment:
Same base types should result in a dimensionless result, not a measure of the same unit